ecommerce/frontend/src/components/SignUpForm.js

```diff
@@ -131,159 +167,187 @@ export const SignUpForm = ({ onSubmit = () => { } }) => {
-                        message: 'Please select your state!',
-                    },
-                ]}
-            >
-                <Input />
-            </Form.Item>
+            {isVendor && (
+                <Form.Item
+                    name={['address', 'state']}
+                    label="State"
+                    rules={[
+                        {
+                            required: true,
+                            message: 'Please select your state!',
+                        },
+                    ]}>
+                    <Input />
+                </Form.Item>
+            )}
 
-            {/* Zip code input */}
-            <Form.Item
-                name={['address', 'zipCode']}
-                label="Zip Code"
-                rules={[
-                    {
-                        required: true,
-                        message: 'Please enter your zip code!',
-                    },
-                ]}
-            >
-                <Input />
-            </Form.Item>
-            
             {/* Main address input */}
-            <Form.Item
-                name={['address', 'mainAddress']}
-                label="Main Address"
-                rules={[
-                    {
-                        required: true,
-                        message: 'Please enter your address!',
-                    },
-                ]}
-            >
-                <Input />
-            </Form.Item>
+            {isVendor && (
+                <Form.Item
+                    name={['address', 'mainAddress']}
+                    label="Main Address"
+                    rules={[
+                        {
+                            required: true,
+                            message: 'Please enter your address!',
+                        },
+                    ]}>
+                    <Input />
+                </Form.Item>
+            )}
 
             {/* Phone number input */}
-            <Form.Item
-                name="phone"
-                label="Phone Number"
-            >
-                <Input.Group compact >
+            <Form.Item name="phone" label="Phone Number">
+                <Input.Group compact>
                     <Form.Item
                         name={['phone', 'countryCode']}
                         noStyle
-                        rules={[{ required: true, message: 'Please enter your country code!' }]}
-                    >
+                        rules={[
+                            {
+                                required: true,
+                                message: 'Please enter your country code!',
+                            },
+                            {
+                                message: 'Wrong country code format',
+                                validator: (rule, val) =>
+                                    /\+\d{1,3}/.test(val) ? Promise.resolve(true) : Promise.reject("Country code format is incorrect!"),
+                            },
```

Comment:
nitpick: if you have time, make it so that the validator returns an error only when both the field is not empty and the value given does not fit the format. this is done so that you dont get both "please enter country code" and "wrong country code format" when the field is empty. this suggestion is optional, we might leave it for another PR. if you are going to do it, do it for all the similar fields.